src/numerics/petsc_vector.C

```diff
@@ -1151,12 +1151,6 @@ void PetscVector<T>::_get_array(bool read_only) const
 {
   libmesh_assert (this->initialized());
 
-#ifdef LIBMESH_HAVE_CXX11_THREAD
-  std::atomic_thread_fence(std::memory_order_acquire);
-#else
-  Threads::spin_mutex::scoped_lock lock(_petsc_vector_mutex);
```

Comment:
This is sort of separate from this PR, but are we still maintaining code for different thread "models" e.g. TBB and pthreads in the `Threads` namespace? That seems like an increasingly difficult proposition as the thread support in `std::` gets more sophisticated.

Reply:
It does seem to finally be redundant at this point, doesn't it?